### ISSUE TITLE
Feature request: Logon-fail timeout

### ISSUE THREAD

**Issue by timogith** [friendica/friendica#8889]
### Expected behavior
A simple short timeout after a defined number of failed login attempts could prevent simple brute-force attacks. As an alternative, a captcha option could be included.

### Actual behavior
The main login is currently not designed to include an anti-brute-force measure. There is no add-on for such a functionality as well, as far as I can tell. This necessitates the use of really strong passwords or two-factor authentication (which, however, is an obstacle to some). This matter is even worse since user names are usually public.

### Steps to reproduce the problem
Log on frequently, or script an automatized login.

### Friendica version you encountered the problem
2020.07 (applies to all though)

### PHP version
irrelevant

### SQL version
irrelevant

**Comment by MrPetovan** [comment]
Thank you for the suggestion!

**Comment by AlfredSK** [comment]
Yes, a good suggestion. At the moment I use fail2ban to catch brute force login attempts.

**Comment by timogith** [comment]
Thank you.

I have Friendica fail2banned as well. However, even if there were only responsible and sec-savvy admins out there going for anti-brute-force instanter, requiring an additional set-up of an IPS might pose a problem with several hosting providers (or their customers, respectively), and besides, fail2ban has turned out leaving users clueless as to whether the server is down after failing to enter the password thrice or five times or so. Fail2ban does not indicate any reason for what seems to be a sudden breakdown to the user.

Even though I did once write a PHP captcha script (which would do at least a similar trick) ages ago, I am not well-versed in programming, and thus I am afraid I cannot actively contribute to a secure implementation of anything here. I shall be taking a closer look on the translation side instead (quite some dog's breakfast there).

Other ideas, beside the timeout thing, would be (i) the separation of handle and logon name or (ii) introducing some facultative captcha system. Not worth a ticket, though -- I reckon a timeout would be the first and foremost option there.

Thanks a lot for everyone's contributions, by the way. Cheers.

**Comment by AlfredSK** [comment]
I really don't like captchas. And I'm pretty sure they're bad for visually impaired persons. I think it's better to have a built-in brute force protection that disables the login function for some time after x failed attempts.
Hint: Nextcloud has such a protection...

Concerning the fail2ban problem: Yes, it looks like the server is down. That's why I put a hint on the main/login page about the IP block when you fail to login three times. :-) But I agree. It's not a perfect solution.

**Comment by MrPetovan** [comment]
I for one do not want to add a Captcha either. I do not want to separate the handle from the log on identifier either. A cooldown period after several failed attempts is the way to go.
